addon/-private/system/model/internal-model.js

```diff
@@ -889,13 +883,34 @@ export default class InternalModel {
     this.eachRelationship((name, relationship) => {
       if (this._relationships.has(name)) {
         let rel = this._relationships.get(name);
+
         rel.clear();
         rel.removeInverseRelationships();
       }
     });
     Object.keys(this._implicitRelationships).forEach((key) => {
-      this._implicitRelationships[key].clear();
-      this._implicitRelationships[key].removeInverseRelationships();
+      let rel = this._implicitRelationships[key];
+
+      rel.clear();
+      rel.removeInverseRelationships();
+    });
+  }
+
+  /*
+    @method unloadDeletedRecordFromRelationships
+   */
+  unloadDeletedRecordFromRelationships() {
```

Comment:
per our chat out of band, let's rename this to not conflate this case with `unloadRecord`